hack/presubmit.sh

```diff
@@ -0,0 +1,178 @@
+  CRD_TMPS=$(find $CRD_TMP -iname '*.yaml' | sort)
+  for i in $CRD_TMPS; do
+    b=$(basename ${i})
+    dir=${i%.yaml}
+    mkdir -p ${dir}
+    mv $i ${dir}/$b
+    controller-gen schemapatch:manifests=${dir} output:dir=${dir} paths=./pkg/operator/apis/...
+  done
+  
+  # Merge and overwrite crds.yaml. Remove last line so we don't produce
+  # a final empty file that would make repeated runs of this script fail
+  CRD_TMPS=$(find $CRD_TMP -iname '*.yaml' | sort)
+  cat "${SCRIPT_ROOT}"/hack/boilerplate.txt > ${CRD_DIR}/crds-tmp.yaml
+  echo -e "# NOTE: This file is autogenerated.\n" >> ${CRD_DIR}/crds-tmp.yaml
+  sed -s '$a---' $CRD_TMPS | sed -e '$ d' -e '/^#/d' -e '/^$/d' >> ${CRD_DIR}/crds-tmp.yaml
+  cp ${CRD_DIR}/crds-tmp.yaml ${CRD_DIR}/xx-crds.yaml
+  mv ${CRD_DIR}/crds-tmp.yaml ${EXAMPLES_DIR}/setup.yaml
+}
+
+update_docgen() {
+  echo ">>> generating API documentation"
+  
+  which po-docgen || (go get github.com/prometheus-operator/prometheus-operator \
+    && go install -mod=mod github.com/prometheus-operator/prometheus-operator/cmd/po-docgen)
+  mkdir -p doc
+  po-docgen api ./pkg/operator/apis/monitoring/v1alpha1/types.go > doc/api.md
+  sed -i 's/Prometheus Operator/GMP CRDs/g' doc/api.md
+}
+
+combine() {
+  SOURCE_DIR=$1
+  DEST_YAML=$2
+
+  YAMLS=$(find ${SOURCE_DIR} -regextype sed -regex '^.*/[0-9][0-9]-\w.*.yaml$' | sort)
+  cat "${SCRIPT_ROOT}"/hack/boilerplate.txt > $DEST_YAML
+  echo -e "# NOTE: This file is autogenerated.\n" >> $DEST_YAML
+  sed -s '$a---' $YAMLS | sed -e '$ d' -e '/^#/d' -e '/^$/d' >> $DEST_YAML
+}
+
+update_examples() {
+  echo ">>> regenerating example yamls"
+
+  combine ${SCRIPT_ROOT}/cmd/operator/deploy/operator ${SCRIPT_ROOT}/examples/operator.yaml
+  combine ${SCRIPT_ROOT}/cmd/operator/deploy/rule-evaluator ${SCRIPT_ROOT}/examples/rule-evaluator.yaml
+}
+
+run_tests() {
+  go test `go list ${SCRIPT_ROOT}/... | grep -v operator/e2e`
+}
+
+reformat() {
+  go mod tidy && go mod vendor && go fmt ${SCRIPT_ROOT}/...
+}
+
+exit_msg() {
+  echo $1
+  exit 1
+}
+
+run_all() {
+  # As this command can be slow, optimize by only running if there's difference
+  # from the origin/main branch.
+  codegen_diff || update_codegen
+  reformat
+  update_crdgen
+  update_examples
+  update_docgen
+  run_tests
+}
+
+main() {
+  if [[ -z "$@" ]]; then
+    run_all
+  else
+    for opt in "$@"; do
+      case "${opt}" in
+        all)
+          run_all
+          ;;
+        codegen)
+          update_codegen
+          ;;
+        crdgen)
+          update_crdgen
+          ;;
+        diff)
+          git diff -s --exit-code doc go.mod go.sum '*.go' '*.yaml' || \
+            exit_msg "diff found - ensure regenerated code is up-to-date and committed."
+          ;;
+        docgen)
+          update_docgen
+          ;;
+        examples)
+          update_examples
+          ;;
+        format)
+          reformat
+          ;;
+        test)
+          run_tests
+          ;;
+      esac
```

Comment:
This would just exit when an unknown command is provided, right? Worth adding a default error message.